imaging/ml/ml_codelab/breast_density_auto_ml.ipynb

```diff
@@ -1,935 +1,972 @@
+    "\n",
+    "resp, content = http.request(path, method='POST', headers=headers, body=body)\n",
+    "assert resp.status == 200, 'error exporting to JPEG, code: {0}, response: {1}'.format(resp.status, content)\n",
+    "print('Full response:\\n{0}'.format(content.decode()))\n",
+    "\n",
+    "# Record operation_name so we can poll for it later.\n",
+    "response = json.loads(content.decode())\n",
+    "operation_name = response['name']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "pVhp8IdVkVH3"
+   },
+   "source": [
+    "We will use the Operation ID returned from the previous command to poll the status of ExportDicomData. This will take a few minutes to complete so we will wait until completion. When the operation is complete, the operation's *done* field will be set to true.\n",
+    "\n",
+    "Meanwhile, you should be able to observe the JPEG images being added to your Google Cloud Storage bucket."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "G4GE7pzsl-lP"
+   },
+   "outputs": [],
+   "source": [
+    "import time\n",
+    "\n",
+    "def wait_for_operation_completion(path, timeout): \n",
+    "  success = False\n",
+    "  while time.time() < timeout:\n",
+    "    print('Waiting for operation completion...')\n",
+    "    resp, content = http.request(path, method='GET')\n",
+    "    assert resp.status == 200, 'error polling for Operation results, code: {0}, response: {1}'.format(resp.status, content)\n",
+    "    response = json.loads(content.decode())\n",
+    "    if 'done' in response:\n",
+    "      if response['done'] == True and 'error' not in response:\n",
+    "        success = True;\n",
+    "      break\n",
+    "    time.sleep(30)\n",
+    "\n",
+    "  print('Full response:\\n{0}'.format(content))      \n",
+    "  assert success, \"operation did not complete successfully in time limit\"\n",
+    "  print('Success!')\n",
+    "  return response\n",
+    "  \n",
+    "timeout = time.time() + 10*60 # Wait up to 10 minutes.\n",
+    "path = os.path.join(HEALTHCARE_API_URL, operation_name)\n",
+    "_ = wait_for_operation_completion(path, timeout)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "liblamUVyWZw"
+   },
+   "source": [
+    "Next, we will join the training data stored in Google Cloud Storage with the labels in the TCIA website. The output of this step is a [CSV file that is input to AutoML](https://cloud.google.com/vision/automl/docs/prepare). This CSV contains a list of pairs of (IMAGE_PATH, LABEL)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "DgDI4Esw8bXT"
+   },
+   "outputs": [],
+   "source": [
+    "!sudo pip3 install tensorflow==1.14.0 --ignore-installed\n",
```

Comment:
Add a comment to explain why we are doing this/ highlight this.

Reply:
done

Reply:
can you just install the notebook with tensorflow preloaded instead?

Reply:
it's preloaded but different version

Reply:
it's preloaded but different version